doc/development_model.md

```diff
@@ -0,0 +1,113 @@
+# Thrust Branching and Development Model
+
+The following is a description of how the Thrust development teams approaches branching and release tagging. This
+is a living document that will evolve as our process evolves.
+
+Thrust is distributed in three ways:
+
+   * On GitHub.
+   * In the NVIDIA HPC SDK.
+   * In the CUDA Toolkit.
+
+## Trunk Based Development
+
+Thrust uses [trunk based development](https://trunkbaseddevelopment.com). There is a single long-lived
+branch called `master`. Engineers may create branches for feature development. such branches always
+merge into `master`. There are no release branches. Releases are produced by taking a snapshot of
+`master` ("snapping"). After a release has been snapped from `master`, it will never be changed.
+
+## Repositories
+
+As Thrust is developed both on GitHub and internally at NVIDIA, there's three main places where code lives:
+
+   * The Source of Truth, the [public Thrust repository](https://github.com/thrust/thrust), referred to as
+     `github` later in this document.
+   * An internal GitLab repository, referred to as `gitlab` later in this document.
+   * An internal Perforce repository, referred to as `perforce` later in this document.
+
+## Versioning
+
+Thrust has its own versioning system for releases, independent of the versioning scheme of the NVIDIA
+HPC SDK or the CUDA Toolkit.
+
+Today, Thrust version numbers have a specific [semantic meaning](https://semver.org/).
+Releases prior to 1.10.0 largely, but not strictly, followed these semantic meanings.
+
+The version number for a Thrust release uses the following format:
+`MMM.mmm.ss-ppp`, where:
+
+   * `THRUST_VERSION_MAJOR`/`MMM`: Major version, up to 3 decimal digits. It is incremented
+     when the fundamental nature of the library evolves, leading to widespread changes across the
+     entire library interface with no guarantee of API, ABI, or semantic compatibility with former
+     versions.
+   * `THRUST_VERISON_MINOR`/`mmm`: Minor version, up to 3 decimal digits. It is incremented when
+     breaking API, ABI, or semantic changes with are made.
```

Comment:
The grammar of this sentence is wrong, "or semantic changes with are made".  I don't know what was intended.

Reply:
Yes those are bad words they will be punished.